module/zfs/zvol.c

```diff
@@ -702,21 +742,30 @@ zvol_log_truncate(zvol_state_t *zv, dmu_tx_t *tx, uint64_t off, uint64_t len,
 	zil_itx_assign(zilog, itx, tx);
 }
 
-static int
-zvol_discard(struct bio *bio)
+static void
+zvol_discard(void *arg)
 {
+	struct bio *bio = arg;
 	zvol_state_t *zv = bio->bi_bdev->bd_disk->private_data;
 	uint64_t start = BIO_BI_SECTOR(bio) << 9;
 	uint64_t size = BIO_BI_SIZE(bio);
 	uint64_t end = start + size;
-	int error;
+	int error = 0;
 	rl_t *rl;
 	dmu_tx_t *tx;
+#ifdef HAVE_GENERIC_IO_ACCT
+	unsigned long start_jif = jiffies;
```

Comment:
nit: I prefer to see the start time jiffies adjacent to the generic_start_io_acct()
indeed, you can even include the generic_start_io_acct() inside the ifdef, for clarity